### ISSUE TITLE
disown, then pkill leaves defunct process

### ISSUE THREAD

**Issue by santpent** [fish-shell/fish-shell#5342]
Opening any program in the background, for example
```
gedit &
disown
pkill gedit
ps -aux | grep gedit # shows [gedit] <defunct>
```

It also occurs with `killall`, `kill` and `kill -9`. You then need to close that terminal instance, or kill the parent process, fish.

This doesn't happen with fish 2.7.1, or with bash.

Using the latest nightly, 2.7.1-1694-gab0f1d33 on Ubuntu 18.04.

**Comment by mqudsi** [comment]
This is ugly. The problem is that the parent process on record is still fish, but we don't call waitpid for processes/groups that aren't associated with any jobs, so the kernel keeps the zombie reference around.

We would need to keep a reference to that p(g)id around and always wait on it with `WNOHANG` (in reality, change `remove_job` from actually deleting a job from the job list to simply marking it hidden), start, or else spawn a background thread that does nothing but make a blocking call to `waitpid` to make sure the zombie process gets reaped. Or we could leave it as-is. Yuck.

**Comment by ridiculousfish** [comment]
This is a regression, we should fix it. IMO it should be simple as just maintaining a list of disowned pids, and polling those.

**Comment by mqudsi** [comment]
Closed in  8730b48.
